scylla/src/transport/session_test.rs

```diff
@@ -2206,3 +2206,38 @@ async fn test_prepare_batch() {
         assert!(session.prepare_batch(&bad_batch).await.is_err());
     }
 }
+
+#[tokio::test]
+async fn test_refresh_metadata_after_schema_agreement() {
+    let uri = std::env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9042".to_string());
+    let session = SessionBuilder::new().known_node(uri).build().await.unwrap();
+
+    let ks = unique_keyspace_name();
+    session.query(format!("CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = {{'class' : 'SimpleStrategy', 'replication_factor' : 1}}", ks), &[]).await.unwrap();
+    session.use_keyspace(ks.clone(), false).await.unwrap();
+
+    session
+        .query(
+            "CREATE TYPE IF NOT EXISTS udt (field1 int, field2 uuid, field3 text)",
```

Comment:
Nit: it would perhaps be slightly better no to use `IF NOT EXISTS` here, it would express the intention of the test better. The UDT is not supposed to be in the initial metadata, it should appear only after the statement is executed and the metadata is refreshed.